src/number_theory/primorial.lean

```diff
@@ -0,0 +1,190 @@
+/-
+Copyright (c) 2020 Patrick Stevens. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Patrick Stevens
+-/
+import tactic
+import data.nat.parity
+import data.nat.choose
+
+/-!
+# Primorial
+
+This file defines the primorial function, and proves a bound on its size.
+
+## Notations
+
+We use the local notation `n#` for the primorial of `n`: that is, the product of the primes less
+than or equal to `n`.
+-/
+
+open finset
+open nat
+open_locale big_operators
+
+/-- The primorial `n#` of `n` is the product of the primes less than or equal to `n`.
+-/
+def primorial (n : ℕ) : ℕ := ∏ p in (filter prime (range (n + 1))), p
+local notation x`#` := primorial x
+
+lemma primorial_succ {n : ℕ} (n_big : 1 < n) (r : n % 2 = 1) : (n + 1)# = n# :=
+begin
+  have not_prime : ¬prime (n + 1),
+    { intros is_prime,
+      cases (prime.eq_two_or_odd is_prime),
+      { linarith, },
+      { exfalso,
+        rw ←not_even_iff at h r,
+        have e : even (n + 1 - n), exact (even_sub (le_of_lt (lt_add_one n))).2 (iff_of_false h r),
+        simp only [nat.add_sub_cancel_left, not_even_one] at e,
+        exact e, }, },
+  apply finset.prod_congr,
+  { rw [@range_succ (n + 1), filter_insert, if_neg not_prime], },
+  { exact λ _ _, rfl, },
+end
+
+private lemma range_contains_upper_bound (n : ℕ) : n ∈ range (n + 1) :=
```

Comment:
```suggestion
lemma self_mem_range_succ (n : ℕ) : n ∈ range (n + 1) :=
```
please move this close to wherever `finset.range` is defined.
Alternatively, you can just inline it at the place where you need it.